src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -171,9 +174,13 @@ trait RetrieveByExternalIds[R, F[_]] extends WithRequestSession[F] with BaseUrl
   def retrieveByExternalIds(externalIds: Seq[String]): F[Seq[R]]
   @SuppressWarnings(Array("org.wartremover.warts.IterableOps"))
   def retrieveByExternalId(externalId: String): F[R] =
-    // The API returns an error causing an exception to be thrown if the item isn't found,
-    // so .head is safe here.
-    requestSession.map(retrieveByExternalIds(Seq(externalId)), (r1: Seq[R]) => r1.head)
+    requestSession.map(
+      retrieveByExternalIds(Seq(externalId)),
+      (r1: Seq[R]) =>
+        r1.headOption.getOrElse(
+          throw new SdkException("Unexpected empty response when retrieving item by ExternalId")
```

Comment:
This change is debatable, but I think in practice it is:
 * just as readable
 * doesn't require supress warning and a comment to explain it
 * if it does go wrong for whatever reason we get a more readable exception.